semgrep-core/src/runner/Parse_with_caching.ml

```diff
@@ -98,7 +98,14 @@ let cache_computation use_parsing_cache version_cur file cache_file_of_file f =
 
 let cache_file_of_file parsing_cache_dir suffix filename =
   let dir = parsing_cache_dir in
-  if not (Sys.file_exists dir) then Unix.mkdir dir 0o700;
+  (if not (Sys.file_exists dir) then
+   try
+     Unix.mkdir dir 0o700
+     (* bugfix: Despite the check above, we may race with another thread to
+      * create the directory, leading to a crash if this exception is unhandled.
+      * *)
+   with
+   | Unix.Unix_error (Unix.EEXIST, _, _) -> ());
```

Comment:
whats the consequence of not creating this directory? does this just mean we won't cache in this particular instance?

Reply:
We need it to exist, we don't need to create it ourselves. That's why we check if it exists before trying to create it. The `EEXIST` error just means that we are trying to make something that already exists.

Reply:
oh oops, that's smart. i did not think about what the error actually meant.